csrc/scheduler/vectorize_helper.cpp

```diff
@@ -400,44 +375,56 @@ std::vector<IterDomain*> ContiguousInnerDimensionsMapper::projectId(
     if (it == frontier.end()) {
       return;
     }
-
     auto pos = std::distance(frontier.begin(), it);
-    if (resize_in_pad_.count(resize_op) != 0) {
-      // resize created by PadOp.
 
-      // project resize op to frontier.
-      frontier[pos] = id_to;
-      // clear left of resize, since those are no long contiguous.
-      frontier.erase(frontier.begin(), it);
+    // project resize op to frontier.
+    frontier[pos] = id_to;
+    // clear left of resize, since those are no long contiguous.
+    frontier.erase(frontier.begin(), it);
 
-      if (recording_) {
-        // TODO: support negative resize extent.
-        //
-        // Limit current support to only positive resize extent for now. So we
-        // only consider the pad_extent, which becomes the real buffer on
-        // output. Hence we do GCD among padded extent as well as extent of the
-        // id_from. Note since we are taking the GCD here, I don't think using
-        // id_from or id_to makes a difference.
-        auto consumer_factor = getProjectedExtent(id_from);
-        auto comp = [](Val* factor, Val* extent) {
-          return SimplifyingIrBuilder::whereExpr(
-              SimplifyingIrBuilder::eqExpr(
-                  extent, extent->container()->zeroVal()),
-              factor,
-              // for extent < 0, we'll take max(1, extent). Because of the gcd,
-              // This is effectively excluding the resize id from vectorization.
-              SimplifyingIrBuilder::gcdExpr(
-                  factor,
-                  SimplifyingIrBuilder::maxExpr(
-                      extent->container()->oneVal(), extent)));
-        };
-        consumer_factor = comp(consumer_factor, resize_op->leftExpand());
-        consumer_factor = comp(consumer_factor, resize_op->rightExpand());
-        addProjectedExtent(id_to, consumer_factor);
+    if (recording_) {
+      // we need to check slice offset at this point.
+      auto projected_extent = getProjectedExtent(id_from);
+
+      std::vector<Val*> expands = {
+          resize_op->leftExpand(), resize_op->rightExpand()};
+      if (!p2c) {
+        // reverse pad extent for c2p propagation
+        std::for_each(expands.begin(), expands.end(), [](Val*& val) {
+          val = SimplifyingIrBuilder::negExpr(val);
+        });
       }
-    } else {
-      // unsupproted resize.
-      frontier.erase(frontier.begin(), it + 1);
+
+      // resize extent == 0: no resizing, return the factor as-is
+      // resize extent  > 0: padding, take gcd(factor, extent)
+      // resize extent  < 0: slicing, we ignore the resize when slicing on right
+      // (return factor as-is), but we take gcd(abs(factor), extent) when
+      // slicing on left. This is a conservative analysis of the offset for data
+      // accessing. A better approach needs to consider the actual start pointer
+      // address and handle it in alignment analysis in runtime info.
```

Comment:
The handling of the negative extent here is our conservative handling of offset check. We do not consider the actual offset of the tensor. We would miss out on opportunities where slice could have returned an offset valid for vectorization, when neither the tensor ptr offset nor the extent here are legit for vectorization.

Reply:
offline comment from @naoyam , how about stacked resize, is it safe to ignore the negative right-side resize extent.

Reply:
Update as suggested.